query.c

```diff
@@ -1297,6 +1297,42 @@ answer_lookup_zone(struct nsd *nsd, struct query *q, answer_type *answer,
 		return;
 	}
 	assert(closest_encloser); /* otherwise, no q->zone would be found */
+	if(q->zone->opts && q->zone->opts->pattern
+	&& q->zone->opts->pattern->allow_query) {
+		struct acl_options *why = NULL;
+
+		/* check if it passes acl */
+		if(acl_check_incoming(
+		   q->zone->opts->pattern->allow_query, q, &why) != -1) {
+			assert(why);
+			DEBUG(DEBUG_QUERY,1, (LOG_INFO, "query %s passed acl %s %s",
+				dname_to_string(q->qname, NULL),
+				why->ip_address_spec,
+				why->nokey?"NOKEY":
+				(why->blocked?"BLOCKED":why->key_name)));
+		} else { 
+			if (verbosity >= 2) {
+				char address[128];
+				addr2str(&q->addr, address, sizeof(address));
+				VERBOSITY(2, (LOG_INFO, "query %s from %s refused, %s%s",
```

Comment:
I thought you would want a space between the two explanatory strings.

```suggestion
				VERBOSITY(2, (LOG_INFO, "query %s from %s refused, %s %s",
```